solver/src/settlement_submission/flashbots_api.rs

```diff
@@ -99,4 +118,125 @@ impl FlashbotsApi {
         ensure!(status.is_success(), "status {}: {:?}", status, body);
         Ok(())
     }
+
+    /// Query gas_price for the current network state (simplest one for Flashbots)
+    pub async fn gas_price(&self) -> Result<EstimatedGasPrice> {
+        let body = serde_json::json!({
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "eth_gasFees",
+            "params": [],
+        });
+        let response = self.client.post(URL).json(&body).send().await?;
+        let status = response.status();
+        let body = response.text().await?;
+        ensure!(status.is_success(), "status {}: {:?}", status, body);
+
+        match serde_json::from_str::<jsonrpc_core::Output>(&body) {
+            Ok(body) => match body {
+                jsonrpc_core::Output::Success(body) => {
+                    match serde_json::from_value::<FlashbotGasPrice>(body.result) {
+                        Ok(gas_price) => Ok(EstimatedGasPrice {
+                            eip1559: Some(GasPrice1559 {
+                                base_fee_per_gas: match u64::from_str_radix(
+                                    &gas_price.base_fee_per_gas[2..],
+                                    16,
+                                ) {
+                                    Ok(base_fee) => base_fee as f64,
+                                    Err(err) => {
+                                        return Err(anyhow!(
+                                            "failed to parse base_fee_per_gas: {}",
+                                            err
+                                        ))
+                                    }
+                                },
+                                max_fee_per_gas: match u64::from_str_radix(
+                                    &gas_price.default.max_fee_per_gas[2..],
+                                    16,
+                                ) {
```

Comment:
Parsing u64 from a hex string starting with 0x is repeated several times so would be nice to have that as it's own function

Reply:
If we apply the suggestion :point_up: to use `U256` instead of `String` for the `FlashBotGasPrice` and `Eip1559` types, then we can use `to_f64_lossy()` here to parse `0x...` integers into `f64`s.

Reply:
Used `to_f64_lossy()`